### PR TITLE
Updated Split-Path to work with UNC Roots

### PR DESCRIPTION
**Pull request by dchristian3188**
Currently splitting a UNC root on windows will return a empty string.

Please reference:
#2301

### CODE REVIEW

File: test/powershell/Provider/FileSystemProvider.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Describe "SplitPath Tests (Windows Only)" -tags "CI" {
```
Comment (JamesWTruher) [inline test/powershell/Provider/FileSystemProvider.Tests.ps1]: >Windows  [](start = 27, length = 8)

It seems like these should just be added to `test/powershell/modules/Microsoft.PowerShell.Management/Split-Path.Tests.ps1`

Comment (JamesWTruher) [inline test/powershell/Provider/FileSystemProvider.Tests.ps1]: thanks!

File: test/powershell/Modules/Microsoft.PowerShell.Management/Split-Path.Tests.ps1 [hunk 2]
```diff
@@ -69,4 +69,53 @@ Describe "Split-Path" -Tags "CI" {
 	Split-Path -Parent "/usr/local/bin"  | Should Be "${dirSep}usr${dirSep}local"
 	Split-Path -Parent "usr/local/bin"   | Should Be "usr${dirSep}local"
     }
+
+    Context "SplitPath Tests (Windows Only)" {
```
Comment (vors) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Split-Path.Tests.ps1]: Why it's Windows Only? Can we cover all systems in these tests? I think the attitude of https://github.com/PowerShell/PowerShell/issues/2301#issuecomment-261118596 is to provide a consistent experience on all platforms.

**Review by vors — CHANGES_REQUESTED** [review]
Great work! Can we make tests more generic to cover Windows and non-Windows platforms?

**Comment by vors** [comment]
Waiting on non-windows tests.

**Comment by dchristian3188** [comment]
Updated Tests. Please let me know if everything looks ok

**Review by vors — APPROVED** [review]

**Pull request merged**